### PR TITLE
Update process for defining target platform

### PR DESCRIPTION
**Pull request by briandealwis**

### CODE REVIEW

**Comment by chanseokoh** [comment]
I wonder if we want to delete `eclipse/ide-target-platform/ide-target-platform.target`.

File: README.md [hunk 1]
```diff
@@ -17,7 +17,8 @@ Maven for building Eclipse bundles and features.
   this somewhere on your file system.
 
 1. [Eclipse 4.5 (Mars) or later](https://www.eclipse.org/downloads/eclipse-packages/). 
-  It's easiest to use the _Eclipse IDE for Java EE Developers_ package.
+  It's easiest to use the _Eclipse IDE for Java EE Developers_ package. You can use
+  Eclipse 4.6 (Neon) or 4.7 (Oxygen) as we define a target platforms to build against.
```
Comment (elharo) [inline README.md]: "a target platform" (not plural)

File: README.md [hunk 2]
```diff
@@ -147,13 +148,33 @@ described below.
...
+  
+  3. Name it `GCP IDE Target Platform`.
+  
+  4. Select `Add` > `Software Site`.
+  
+  5. Select the `Add...` button (found beside the `Work with:` field) and then select `Local`
+     to find a local repository. Navigate to `.../eclipse/ide-target-platform/target/repository`,
+     and click `OK`.
+     
+  6. Once the main content populates, tick the `Uncategorized` item to pull in all items. Click `Finish`.
```
Comment (elharo) [inline README.md]: tick --> check

File: README.md [hunk 3]
```diff
@@ -147,13 +148,33 @@ described below.
...
+  
+  7. Click `Finish` to complete defining the new target platform definition.
```
Comment (elharo) [inline README.md]: complete defining --> complete

**Comment by briandealwis** [comment]
Thanks for the reminder @chanseokoh — I meant to do that.

File: eclipse/ide-target-platform/ide-target-platform.target [hunk 4]
```diff
@@ -1,6 +0,0 @@
-<?xml version="1.0" encoding="UTF-8" standalone="no"?>
-<?pde version="3.8"?><target name="GCP Development Target (ide-target-repository)" sequenceNumber="18">
-<locations>
-<location path="${project_loc:/ide-target-platform.repo}/target/repository/plugins" type="Directory"/>
```
Comment (briandealwis) [inline eclipse/ide-target-platform/ide-target-platform.target]: Unfortunately the p2-repository location doesn't support variables.

Comment (briandealwis) [inline eclipse/ide-target-platform/ide-target-platform.target]: …so we can't just use a common copy like we did with the `ide-target-platform.target`

File: README.md [hunk 5]
```diff
@@ -4,7 +4,7 @@
 This project provides an Eclipse plugin for building, debugging, and deploying Google Cloud Platform applications.
 
 _TL;DR_: `mvn -Dtycho.toolchains=SYSTEM package` should
-generate a build into `gcp-repo/target/repository`.
+generate a p2-accessible repository in `gcp-repo/target/repository`.
```
Comment (akerekes) [inline README.md]: `gcp-repo/target/repository` should be `ide-target-platform/target/repository`

File: README.md [hunk 6]
```diff
@@ -17,7 +17,8 @@ Maven for building Eclipse bundles and features.
   this somewhere on your file system.
 
 1. [Eclipse 4.5 (Mars) or later](https://www.eclipse.org/downloads/eclipse-packages/). 
-  It's easiest to use the _Eclipse IDE for Java EE Developers_ package.
+  It's easiest to use the _Eclipse IDE for Java EE Developers_ package. You can use
+  Eclipse 4.6 (Neon) or 4.7 (Oxygen) as we define a target platform to build against.
```
Comment (akerekes) [inline README.md]: I thought we're supporting Mars and Neon currently. I don't see Oxygen as target in the root pom.

File: README.md (hunk 5 above)
Comment (briandealwis) [inline README.md]: That's not correct: `gcp-repo` is for our built artifacts.  `ide-target-platform` is for our development use.

File: README.md (hunk 6 above)
Comment (briandealwis) [inline README.md]: You can develop using any version of Eclipse >= what is in your target platform.

Comment (briandealwis) [inline README.md]: Maybe I'll reword that 'to build _and run_ against'.

**Comment by briandealwis** [comment]
PTAL @akerekes — I also made a couple of other tweaks.

**Pull request merged**